src/rgw/rgw_reshard.cc

```diff
@@ -792,8 +958,24 @@ int RGWReshard::process_single_logshard(int logshard_num)
     ldout(store->ctx(), 5) << __func__ << "(): failed to acquire lock on " << logshard_oid << dendl;
     return ret;
   }
-
-  utime_t lock_start_time = ceph_clock_now();
+  Clock::time_point lock_start_time = Clock::now();
+  const char* const log_func_name = __func__;
+
+  auto renew_locks_callback =
+    [&l, &lock_start_time, &logshard_oid, &log_func_name, this](const Clock::time_point& now) -> int {
```

Comment:
i really like the RGWBucketReshardLock type - could we use that the shard locks too? if the constructor takes reshard_oid directly instead of bucket_info, you could just initialize one with logshard_oid here

then instead of RenewLocksCallback, RGWBucketReshard could just take an optional pointer to RGWBucketReshardLock